simulator/mips/mips_instr.cpp

```diff
@@ -14,151 +14,162 @@
 
 #include "mips_instr.h"
 
-const FuncInstr::ISAEntry FuncInstr::isaTable[] =
+//unordered map for R-instructions
+const std::unordered_map <uint8, FuncInstr::ISAEntry> isaMapR =
 {
-    { "###", 0xFF, FORMAT_UNKNOWN, OUT_UNKNOWN, 0, &FuncInstr::execute_unknown, 1},
+    {0xFF, {"###", 0xFF, FORMAT_UNKNOWN, OUT_UNKNOWN, 0, &FuncInstr::execute_unknown, 1} } ,
 
-    // **************** R INSTRUCTIONS ****************
+     // **************** R INSTRUCTIONS ****************
     // Constant shifts
-    // name funct   format    operation memsize           pointer
-    { "sll", 0x0, FORMAT_R, OUT_R_SHAMT, 0, &FuncInstr::execute_sll, 1},
+    //   key        name  funct    format operation  memsize           pointer
+    {0x0, { "sll" , 0x0, FORMAT_R, OUT_R_SHAMT, 0, &FuncInstr::execute_sll, 1} },
```

Comment:
Do we need to have functor/opcode inside ISAEntry now?

Reply:
No, it's not obligatory